src/modules/tccpp/components/utility-tools.ts

```diff
@@ -32,7 +32,7 @@ export default class UtilityTools extends BotComponent {
         if (message.author.bot) {
             return;
         }
-        if (this.wheatley.is_authorized_mod(message.author)) {
+        if (await this.wheatley.fetch_member_if_permitted(message.author, Discord.PermissionFlagsBits.ManageChannels)) {
```

Comment:
Thoughts on making the pattern instead be this?
```ts
if (await this.wheatley.check_permissions(message.author, Discord.PermissionFlagsBits.ManageChannels)) {
```

I think `fetch_member_if_permitted` is a clever way of making a single utility that can be used for a few things but it would be more clear to just have a check permission function. The current vocabulary leaves open the question "why wouldn't it be permitted [for wheatley?] to fetch a member"

Reply:
Yeah I was not sure about the name either. I think I ultimately only ended up using the member result in one place, so we could also just change it to a simple boolean check. Or maybe keep the member lookup feature and name it `with_permissions`?

Reply:
I think it would be better over all to just have two functions here, the one that fetches and does a check can use the first method's check logic as well. I'll defer to you though :)

Reply:
well, free enough anyways ^^